src/auto-launch.js

```diff
@@ -1,29 +1,86 @@
 import AutoLaunch from 'auto-launch'
+import { app } from 'electron'
+import os from 'os'
+import path from 'path'
+import fs from 'fs-extra'
+import untildify from 'untildify'
 import createToggler from './create-toggler'
 import logger from './common/logger'
 import store from './common/store'
 
 const CONFIG_KEY = 'autoLaunch'
 
-const autoLauncher = new AutoLaunch({
-  name: 'IPFS Desktop'
-})
+function isSupported () {
+  const plat = os.platform()
+  return plat === 'linux' || plat === 'win32' || plat === 'darwin'
+}
+
+// Disable the old auto launch mechanism.
+// TODO: remove on 0.10.0.
+async function disableOldLogin () {
+  try {
+    const autoLauncher = new AutoLaunch({ name: 'IPFS Desktop' })
+
+    if (await autoLauncher.isEnabled()) {
+      await autoLauncher.disable()
+      logger.error('[launch on startup] old mechanism disabled')
+    }
+  } catch (_) {
+    // ignore...
+  }
+}
+
+function getDesktopFile () {
+  return path.join(untildify('~/.config/autostart/'), 'ipfs-desktop.desktop')
+}
+
+async function enable () {
+  if (app.setLoginItemSettings) {
+    app.setLoginItemSettings({ openAtLogin: true })
+  }
```

Comment:
To be sure there are no duplicated autoruns when `setLoginItemSettings` adds support for Linux, we could run this only on macos and windows, and finish immediately on those platforms, so useless `.desktop` is not created on them (same with `disable`):

```js
  if (app.setLoginItemSettings && ['win32', 'darwin'].includes(os.platform())) {
    app.setLoginItemSettings({ openAtLogin: true })
    return
  }
```

Reply:
Thanks. Completely forgot the `return`. However, I don't think adding the second verification on the `if` is necessary since the function only exists on supported platforms.